### ISSUE TITLE
Fix false positives for dollar variables in `no-invalid-position-at-import-rule`

### ISSUE THREAD

**Issue by XhmikosR** [stylelint/stylelint#5263]
> Clearly describe the bug

Should allow dollar variables

> Which rule, if any, is the bug related to?

e.g. `no-invalid-position-at-import-rule`

> What code is needed to reproduce the bug?

e.g.

```scss
/*!
 * Bootstrap Reboot v5.0.0-beta3 (https://getbootstrap.com/)
 * Copyright 2011-2021 The Bootstrap Authors
 * Copyright 2011-2021 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE)
 * Forked from Normalize.css, licensed MIT (https://github.com/necolas/normalize.css/blob/master/LICENSE.md)
 */

@import "functions";
@import "variables";
// Prevent the usage of custom properties since we don't add them to `:root` in reboot
$font-family-base: $font-family-sans-serif; // stylelint-disable-line scss/dollar-variable-default
$font-family-code: $font-family-monospace; // stylelint-disable-line scss/dollar-variable-default
@import "mixins";
@import "reboot";
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-invalid-position-at-import-rule": true
  }
}
```

> Which version of stylelint are you using?

e.g. `13.13.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"

> What did you expect to happen?

e.g. "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
bootstrap/scss/bootstrap-reboot.scss
 14:1  ✖  Unexpected invalid position @import rule   no-invalid-position-at-import-rule
 15:1  ✖  Unexpected invalid position @import rule   no-invalid-position-at-import-rule
```

Refs https://github.com/stylelint/stylelint/pull/5202#issuecomment-827602316

**Comment by jeddy3** [comment]
@XhmikosR Thanks for the report and for using the template.

Let's change the logic in the rule so that it only warns if `@import` is preceded by either a standard syntax:

- rule, i.e. `isStandardSyntaxRule(rule)`
- at-rule, i.e. `isStandardSyntaxAtRule(atRule)`

This will be true to the [wording of the spec](https://drafts.csswg.org/css-cascade/#at-import) while allowing non-standard syntax.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
